javascripts/discourse/components/header-search.js

```diff
@@ -1,10 +1,19 @@
 import Component from "@glimmer/component";
 import { service } from "@ember/service";
+import { modifier as modifierFn } from "ember-modifier";
+import { focusSearchInput } from "discourse/components/search-menu";
 
 export default class HeaderSearch extends Component {
   @service site;
   @service siteSettings;
   @service currentUser;
+  @service appEvents;
+
+  handleKeyboardShortcut = modifierFn(() => {
+    const cb = () => focusSearchInput();
```

Comment:
can we pass `focusSearchInput` directly to `on`/`off?`

Reply:
We could, but *theoretically* it would be possible for some other code to do the same, and then the `off` calls would interfere with each other. I realise that's incredibly unlikely, but still I prefer to use a locally-scoped function which no other code can hold a reference to.

Reply:
ah, good point